src/base/static/libs/maps/mapboxgl-provider.js

```diff
@@ -119,32 +119,269 @@ const configRuleToFillLayer = (layerConfig, i) => {
   };
 };
 
+const DEFAULT_STYLE_NAME = "Mapseed default style";
+const defaultStyle = {
+  version: 8,
+  name: DEFAULT_STYLE_NAME,
+  sources: {},
+  layers: [],
+  sprite: `${window.location.protocol}//${
+    window.location.host
+  }/static/css/images/markers/spritesheet`,
+};
+
 export default (container, options) => {
   options.map.container = container;
-  options.map.style = {
-    version: 8,
-    sources: {},
-    layers: [],
-    sprite: `${window.location.protocol}//${
-      window.location.host
-    }/static/css/images/markers/spritesheet`,
-  };
+  options.map.style = defaultStyle;
 
   const layersCache = {};
+  const sourcesCache = {};
+
+  const floatSymbolLayersToTop = () => {
+    // To ensure that point (aka "symbol") geometry is not obscured by
+    // the new layer, we move all symbol layers to the top of the layer
+    // stack.
+    Object.values(layersCache)
+      .reduce((flat, toFlatten) => {
+        return flat.concat(toFlatten);
+      }, [])
+      .filter(internalLayer => internalLayer.id.includes("symbol"))
+      .forEach(internalLayer => map.moveLayer(internalLayer.id));
+  };
+
+  const addInternalLayers = (layerId, isBasemap) => {
```

Comment:
Can we call these something more descriptive than "internal layers"? Perhaps we can call them "mapbox layers", and rename this function "addMapboxLayers"? We may also want to clarify our references to our first class "config layers" by naming them `configLayer` instead of `layer`.

Reply:
👍